static/js/pm_list.js

```diff
@@ -1,52 +1,71 @@
 import $ from "jquery";
+import _ from "lodash";
 
-import * as narrow_state from "./narrow_state";
-import * as people from "./people";
 import * as pm_list_data from "./pm_list_data";
 import * as pm_list_dom from "./pm_list_dom";
+import * as resize from "./resize";
 import * as stream_popover from "./stream_popover";
 import * as ui from "./ui";
 import * as ui_util from "./ui_util";
 import * as vdom from "./vdom";
 
 let prior_dom;
-let private_messages_open = false;
 
 // This module manages the "Private messages" section in the upper
 // left corner of the app.  This was split out from stream_list.js.
 
-function get_filter_li() {
-    return $(".top_left_private_messages .private_messages_header");
+let private_messages_collapsed = false;
+
+// The private messages section can be zoomed in to view more messages.
+// This keeps track of if we're zoomed in or not.
+let zoomed = false;
+
+function get_private_messages_section_header() {
+    return $("#private_messages_section #private_messages_list .private-messages-section-header");
 }
 
-function set_count(count) {
-    ui_util.update_unread_count_in_dom(get_filter_li(), count);
+export function set_count(count) {
+    ui_util.update_unread_count_in_dom(get_private_messages_section_header(), count);
 }
 
 function remove_expanded_private_messages() {
     stream_popover.hide_topic_popover();
-    ui.get_content_element($("#private-container")).empty();
+    // While making the private messages list empty
+    // we check for any active conversation thread present in it and
+    // if yes, then append it below the collapsed PM section.
+    const $active_conversation_thread = $(".expanded_private_messages li.active-sub-filter");
+    $(".expanded_private_messages").empty().append($active_conversation_thread);
 }
```

Comment:
I think this function can be deleted, in favor of its caller calling `update_private_messages` instead. This hackery to try to save a row has no benefit over the common code.

Reply:
We always do not call `update_private_messages` again everywhere; this function gets into action whenever we call for `close()` ; so do you mean to call `update_private_messages` everywhere and delete this function ?>

Reply:
Yes.

Reply:
(In particular, `update_private_messages` already will very efficiently render just the one row in the case that we would have been running this code anyway).